src/providers/martian/chatComplete.ts

```diff
@@ -0,0 +1,230 @@
+import { MARTIAN, ANTHROPIC } from '../../globals';
+import {
+  ChatCompletionResponse,
+  ErrorResponse,
+  ProviderConfig,
+} from '../../types';
+import { MartianErrorResponseTransform } from './utils';
+
+export const MartianChatCompleteConfig: ProviderConfig = {
+  model: {
+    param: 'model',
+    required: true,
+    default: 'router',
+  },
+  messages: {
+    param: 'messages',
+    required: true,
+  },
+  max_tokens: {
+    param: 'max_tokens',
+    default: 100,
+    min: 0,
+  },
+  temperature: {
+    param: 'temperature',
+    default: 1,
+    min: 0,
+    max: 2,
+  },
+  top_p: {
+    param: 'top_p',
+    default: 1,
+    min: 0,
+    max: 1,
+  },
+  n: {
+    param: 'n',
+    default: 1,
+  },
+  stream: {
+    param: 'stream',
+    default: false,
+  },
+  stop: {
+    param: 'stop',
+  },
+  max_cost: {
+    param: 'max_cost',
+  },
+  max_cost_per_million_tokens: {
+    param: 'max_cost_per_million_tokens',
+  },
+  models: {
+    param: 'models',
+  },
+  willingness_to_pay: {
+    param: 'willingness_to_pay',
+  },
+  extra: {
+    param: 'extra',
+  },
+};
+
+export interface MartianChatCompleteResponse extends ChatCompletionResponse {
+  system_fingerprint: string;
+}
+
+export const MartianChatCompleteResponseTransform: (
+  response: MartianChatCompleteResponse | ErrorResponse,
+  responseStatus: number
+) => ChatCompletionResponse | ErrorResponse = (response, responseStatus) => {
+  if (responseStatus !== 200 && 'error' in response) {
+    return MartianErrorResponseTransform(response, MARTIAN);
+  }
+
+  return response;
+};
+
+/**
+ * Transforms an Martian-format chat completions JSON response into an array of formatted Martian compatible text/event-stream chunks.
+ *
+ * @param {Object} response - The MartianChatCompleteResponse object.
+ * @param {string} provider - The provider string.
+ * @returns {Array<string>} - An array of formatted stream chunks.
+ */
+export const MartianChatCompleteJSONToStreamResponseTransform: (
```

Comment:
JSONToStream transformer is only used for OpenAI provider. It is not required to write it for each provider because it assumes that the json that is passed to the function will always be OpenAI compliant. 

Its already handled in `responseHandler` function. So I think it would be safe to remove it from here.